src/test/scala/Demo4Java.java

```diff
@@ -0,0 +1,115 @@
+            + "&tile=5-&tileStitch=ct&"
+            + "&shiftColsSW=-1&shiftRowsSW=1&shiftColsSE=1&shiftRowsSE=1",
+        "patchWidth=12&patchHeight=12" + "&tile=831,4-7,-5-&tileStitch=ct&"
+            + "shiftColsSW=-2&shiftRowsSW=2&shiftColsSE=2&shiftRowsSE=2",
+        "patchWidth=11&patchHeight=12" //
+            + "&tile=B-C-,---5,C-B-,-5--&tileStitch=ct"
+            + "&shiftColsSW=0&shiftRowsSW=4&shiftColsSE=4&shiftRowsSE=4",
+        "patchWidth=7&patchHeight=16" + "&footside=-5,5-&footsideStitch=-"
+            + "&tile=-5-,5-5,-5-,B-C,-5-&tileStitch=ct" + "&headside=5-,-5&headsideStitch=-"
+            + "&shiftColsSW=-2&shiftRowsSW=4&shiftColsSE=2&shiftRowsSE=4",
+        // more examples for testing: see the demo section of the tiles page
+        // TODO fix the next patterns
+        "patchWidth=8&patchHeight=14" // shared start node in top left corner: trouble for droste
+            + "&footside=b,-,a,-&footsideStitch=-" + "&tile=831,4-7,-5-&tileStitch=ctct"
+            + "&shiftColsSW=-2&shiftRowsSW=2&shiftColsSE=2&shiftRowsSE=2",
+        "patchWidth=7&patchHeight=13" + "&footside=-c14,b4--,-7c3,b-48&footsideStitch=-"
+            + "&tile=831,488,-4-&tileStitch=ctc" + "&headside=44-,11C,87-,88C&headsideStitch=-"
+            + "&shiftColsSW=-2&shiftRowsSW=2&shiftColsSE=2&shiftRowsSE=2",
+
+    };
+    for (int i = 0; i <= urlQueries.length - 1; i++)
+      drosteSteps(urlQueries[i], i);
+
+    // squared diagram
+    SheetSVG patterns = new SheetSVG(2, "height='100mm' width='110mm'", "GFP");
+    patterns.add("586- -789 2111 -4-4", "checker");
+    new java.io.FileOutputStream("sheet.svg").write(patterns.toSvgDoc().getBytes());
+  }
+
+  private static void drosteSteps(String urlQuery, int i) throws IOException {
+    System.out.println("-------------- " + i);
+    TilesConfig config = new TilesConfig(urlQuery);
+    new FileOutputStream(dir + "/" + i + "-prototype.svg")
+        .write((D3jsSVG.prolog() + PrototypeDiagram.create(config)).getBytes());
+
+    Diagram pairs = NewPairDiagram.create(config);
+    generateDiagram(i + "-pairs", "1px", pairs, config, 1);
+    Diagram threads = ThreadDiagram.create(pairs);
+    generateDiagram(i + "-threads", "2px", threads, config, 2);
+
+    // for "ctct" alternatives see:
+    // https://d-bl.github.io/GroundForge/help/Choose-Stitches#assign-stitches
+    Diagram drostePairs = PairDiagram.create("ctct", threads);
+    generateDiagram(i + "-droste-pairs", "1px", drostePairs, config, 2);
+    Diagram drosteThreads = ThreadDiagram.create(drostePairs);
+    generateDiagram(i + "-droste-threads", "2px", drosteThreads, config, 4);
+  }
+
+  private static void generateDiagram(String fileName, String strokeWidth, Diagram diagram,
+      TilesConfig config, Integer scale) throws IOException {
+    System.out.println("-------------- " + fileName);
+
+    LinkedNodes[] linkedNodes = config.linksOfCenterTile(diagram, scale);
+    diagram.logTileLinks(linkedNodes);
```

Comment:
@veronika 
[zipped jar](https://github.com/d-bl/GroundForge/files/2730425/GroundForge-0.x-SNAPSHOT.jar.zip) for fde04d71
Order of linkedNodes as well as sources and targets are not yet defined. When the coordinates of sources or targets are identical, you can sort by reversed id-s.

Sample of logged linkedNodes:
<img width="717" alt="linekdnodes" src="https://user-images.githubusercontent.com/10553630/50735692-23442f00-11b3-11e9-9bcc-5005d3b777a8.png">